x-pack/plugins/actions/server/builtin_action_types/email.ts

```diff
@@ -36,6 +36,7 @@ const ConfigSchemaProps = {
   port: schema.nullable(portSchema()),
   secure: schema.nullable(schema.boolean()),
   from: schema.string(),
+  hasAuth: schema.boolean({ defaultValue: true }),
```

Comment:
I think we have an interesting case here of wanting to be able to cross validate secrets (user/pass) and config (hasAuth).  Not sure we've done that before, not quite sure where it would go!  Specifically, if hasAuth is false, user/pass should both be null, and if hasAuth is true, they should both be non-null.

It also doesn't look like the `hasAuth` property is used anywhere on the server-side, but we should probably change the determination of whether to use auth or not, and to validate the three values, here:

https://github.com/elastic/kibana/blob/33d051b73cb12d0c0704df43617d6e9ed1e5c695/x-pack/plugins/actions/server/builtin_action_types/lib/send_email.ts#L56-L61

That probably means throwing a new exception in that code, if the properties aren't consistent.  A function test for that seems possible, but I think we can live with just a jest test for it, in the `builtin_action_types/email.test.ts` module